Source/DafnyLanguageServer/Workspace/Compilation.cs

```diff
@@ -241,90 +241,92 @@ CompilationInput input
+            new BatchCompleted(null /* unused */, result)));
 #pragma warning restore CS8625 // Cannot convert null literal to non-nullable reference type.
-              }
+        }
 
-              updates.OnNext(new BoogieUpdate(canVerify,
-                task,
-                completedCache));
-            }
+        updates.OnNext(new BoogieUpdate(canVerify,
+          task,
+          completedCache));
+      }
 
-            StatusUpdateHandlerFinally();
-            return;
-          }
+      StatusUpdateHandlerFinally();
+      return;
+    }
 
-          var incrementedJobs = Interlocked.Increment(ref runningVerificationJobs);
-          logger.LogDebug(
-            $"Incremented jobs for task, remaining jobs {incrementedJobs}, {Input.Uri} version {Input.Version}");
-
-          statusUpdates.Subscribe(
-            update => {
-              try {
-                HandleStatusUpdate(canVerify, task, update);
-              } catch (Exception e) {
-                logger.LogError(e, "Caught exception in statusUpdates OnNext.");
-              }
-            },
-            e => {
-              if (e is not OperationCanceledException) {
-                logger.LogError(e, $"Caught error in statusUpdates observable.");
-              }
-
-              StatusUpdateHandlerFinally();
-            },
-            StatusUpdateHandlerFinally
-          );
-        }
+    var incrementedJobs = Interlocked.Increment(ref runningVerificationJobs);
+    logger.LogDebug(
+      $"Incremented jobs for task, remaining jobs {incrementedJobs}, {Input.Uri} version {Input.Version}");
 
-        void StatusUpdateHandlerFinally() {
-          try {
-            var remainingJobs = Interlocked.Decrement(ref runningVerificationJobs);
-            logger.LogDebug(
-              $"StatusUpdateHandlerFinally called, remaining jobs {remainingJobs}, {Input.Uri} version {Input.Version}, " +
-              $"startingCompilation.version {Input.Version}.");
-            if (remainingJobs == 0) {
-              MarkVerificationFinished();
-            }
-          } catch (Exception e) {
-            logger.LogCritical(e, "Caught exception while handling finally code of statusUpdates handler.");
-          }
+    statusUpdates.Subscribe(
+      update => {
+        try {
+          HandleStatusUpdate(canVerify, task, update);
+        } catch (Exception e) {
+          logger.LogError(e, "Caught exception in statusUpdates OnNext.");
+        }
+      },
+      e => {
+        if (e is not OperationCanceledException) {
+          logger.LogError(e, $"Caught error in statusUpdates observable.");
         }
-      }
 
-      DecrementJobs();
-    }
-    finally {
-      verificationTickets.Enqueue(Unit.Default);
+        StatusUpdateHandlerFinally();
+      },
+      StatusUpdateHandlerFinally
+    );
+  }
+
+  private void StatusUpdateHandlerFinally() {
+    try {
+      var remainingJobs = Interlocked.Decrement(ref runningVerificationJobs);
```

Comment:
A statement not expected to create an exception should generally not be in the try portion of a try catch statement if possible.

Reply:
I have removed this method in a branch I will publish after this PR is merged